tir/authorizationClient.go

```diff
@@ -97,11 +104,13 @@ func (ac AuthorizingHttpClient) handleAuthorization(tirAddress string) (bearerTo
 		return bearerToken, err
 	}
 
-	metaData, err := ac.getMetaData(tirAddress)
-	if err != nil {
+	metaDataInterface, hit := common.GlobalCache.IssuersCache.Get(tirAddress)
+	if !hit {
 		logging.Log().Warnf("Was not able to get the openid metadata. Err: %v", err)
 		return bearerToken, err
```

Comment:
The place that calls the handleAuth function probably logs the error too. What would you say about wrapping the error here with some more context and return it then?
`fmt.Errorf("Was not able to get the openid metadata from address %s. Err: %w", tirAddress, err)`

Reply:
Yes, good idea